tools/failures.py

```diff
@@ -268,9 +277,52 @@ def analyze_failures(start_date, end_date, testmode, ignore_failure, method):
 
     communicate(failures, to_insert, total_detected, testmode, end_date)
 
+
+# TODO we need add test for this function to make sure it works under any situation
+def update_runnableapi():
+    """
+    Use it to update runnablejobs.json file.
+    """
+    url = "https://index.taskcluster.net/v1/task/gecko.v2.%s.latest.firefox.decision/"
+    latesttask = retry(requests.get, args=(url % "mozilla-inbound", ),
+                       kwargs={'headers': {'accept-encoding': 'json'}, 'verify': True}).json()
+    taskid = latesttask['taskId']
+
+    # The format of expires is like 2017-07-04T22:13:23.248Z and we only want 2017-07-04 part
+    expires = latesttask['expires'].split('T')[0]
+    timetuple = datetime.datetime.strptime(expires, "%Y-%m-%d").timetuple()
+    newtimestamp = time.mktime(timetuple)
+    path = ROOT_DIR + '/runablejobs.json'
+    # we do nothing if the timestamp of runablejobs.json is equal with the latest task
+    # otherwise we download and update it
+    if os.path.isfile(path):
+        with open(path, 'r+') as data:
+            # read the timesstamp of this task from json file
+            oldtime = json.loads(data.read())['meta']['timetamp']
+        if oldtime == newtimestamp:
+            print "The runnable json file is latest already."
+            return
+        else:
+            print "It's going to update your runnable jobs data."
+            download_runnablejob(newtimestamp, taskid)
+    else:
+        print "It's going to help you download the runnable jobs file."
+        download_runnablejob(newtimestamp, taskid)
+
+
+def download_runnablejob(newtimestamp, taskid=None):
+    if taskid:
+        url = "https://treeherder.mozilla.org/api/" \
+              "project/mozilla-inbound/runnable_jobs/?decisionTaskID=%s&format=json" % taskid
+        print url
+        data = retry(requests.get, args=(url, ), kwargs={'headers': headers}).json()
+        data['meta'].update({'timetamp': newtimestamp})
```

Comment:
what do we do if there is no data returned or errors?  I would like to ensure that we don't write a bad runnable.json file, maybe a quick check of the data to make sure there are at least 100 possibly jobs or something to verify before writing the file?